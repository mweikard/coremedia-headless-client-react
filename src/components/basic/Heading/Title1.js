// @flow
import * as React from 'react';
import PropTypes from 'prop-types';

import withStyles from '../../styles/withStyles';

type Props = {
  className: string,
  children: React.Node,
};
const Title1 = ({ className, children }: Props) => <h1 className={className}>{children}</h1>;

Title1.displayName = 'BasicTitle1';

Title1.propTypes = {
  className: PropTypes.string.isRequired,
  children: PropTypes.node,
};

const styles = (theme, props) => theme.typography.title1;

export default withStyles(styles, 'BasicTitle1')(Title1);
