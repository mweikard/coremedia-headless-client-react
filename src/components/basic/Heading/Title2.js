// @flow
import * as React from 'react';
import PropTypes from 'prop-types';

import withStyles from '../../styles/withStyles';

type Props = {
  className: string,
  children: React.Node,
};
const Title2 = ({ className, children }: Props) => <h2 className={className}>{children}</h2>;

Title2.displayName = 'BasicTitle2';

Title2.propTypes = {
  className: PropTypes.string.isRequired,
  children: PropTypes.node,
};

const styles = (theme, props) => theme.typography.title2;

export default withStyles(styles, 'BasicTitle2')(Title2);
