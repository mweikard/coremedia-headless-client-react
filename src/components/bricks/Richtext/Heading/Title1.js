// @flow
import { Title1 } from '../../../basic/Heading';
import withStyles from '../../../styles/withStyles';

const styles = {
  marginTop: '1em',
};

export default withStyles(styles, 'RichtextTitle1')(Title1);
