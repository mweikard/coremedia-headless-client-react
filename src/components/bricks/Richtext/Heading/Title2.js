// @flow
import { Title2 } from '../../../basic/Heading';
import withStyles from '../../../styles/withStyles';

const styles = {
  marginTop: '0.75em',
};

export default withStyles(styles, 'RichtextTitle2')(Title2);
