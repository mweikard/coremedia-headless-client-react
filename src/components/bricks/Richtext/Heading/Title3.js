// @flow
import { Title3 } from '../../../basic/Heading';
import withStyles from '../../../styles/withStyles';

const styles = {
  marginTop: '0.5em',
};

export default withStyles(styles, 'RichtextTitle3')(Title3);
