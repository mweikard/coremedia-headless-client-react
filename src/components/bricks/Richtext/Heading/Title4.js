// @flow
import { Title4 } from '../../../basic/Heading';
import withStyles from '../../../styles/withStyles';

const styles = {
  marginTop: '0.25em',
};

export default withStyles(styles, 'RichtextTitle4')(Title4);
