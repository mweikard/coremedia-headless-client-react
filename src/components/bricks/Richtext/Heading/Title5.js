// @flow
import { Title5 } from '../../../basic/Heading';
import withStyles from '../../../styles/withStyles';

const styles = {};

export default withStyles(styles, 'RichtextTitle5')(Title5);
