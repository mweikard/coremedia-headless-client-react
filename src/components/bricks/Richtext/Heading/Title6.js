// @flow
import { Title6 } from '../../../basic/Heading';
import withStyles from '../../../styles/withStyles';

const styles = {};

export default withStyles(styles, 'RichtextTitle6')(Title6);
