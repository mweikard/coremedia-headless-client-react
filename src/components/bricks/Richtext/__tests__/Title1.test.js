// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { Title1 } from '../Heading';

describe('Title1 Component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Title1>Content</Title1>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
