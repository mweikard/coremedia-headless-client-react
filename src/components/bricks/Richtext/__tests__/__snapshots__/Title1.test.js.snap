// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title1 Component should render correctly 1`] = `
.c0 {
  font-family: Helvetica,Arial,sans-serif;
  font-size: 6rem;
  font-weight: 300;
  text-transform: uppercase;
  line-height: 1.6em;
  color: #000;
  margin-top: 1em;
}

<h1
  className="c0"
>
  Content
</h1>
`;
